buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -724,5 +724,27 @@ protected void memoryCopy(ByteBuffer src, int srcOffset, ByteBuffer dst, int dst
                 dst.put(src);
             }
         }
+
+        @Override
+        protected void finalize() throws Throwable {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Do not override the Object.finalize() method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AObjectFinalizeOverridenCheck)